src/module/rolls/roll.ts

```diff
@@ -41,12 +41,45 @@ export const DIE_LOWEST_FACE = 1
 export const ACTION_DIE_STR = `d${ACTION_DIE_SIDES}`
 export const CHALLENGE_DIE_STR = `d${CHALLENGE_DIE_SIDES}`
 
+/**
+ * Enumerates Ironsworn roll outcomes.
+ * The key corresponds to the i18n key that labels this outcome.
+ * The value is equal to the number of challenge dice beaten by the action score or progress score.
+ */
 export enum ROLL_OUTCOME {
-  MISS = 'Miss',
-  WEAK_HIT = 'Weak_hit',
-  STRONG_HIT = 'Strong_hit',
+  /**
+   * **Miss:** The score beats neither challenge die.
+   */
+  Miss = 0,
+  /**
+   * **Weak hit:** The score beats one challenge die.
+   */
+  Weak_hit = 1,
+  /**
+   * **Strong hit:** The score beats both challenge dice.
+   */
+  Strong_hit = 2,
+}
+
+/**
+ * Enumerates Dataforged move outcome keys using the same values as {@link ROLL_OUTCOME}
+ */
+export enum DfRollOutcome {
+  /**
+   * **Miss:** The score beats neither challenge die.
+   */
+  Miss = 0,
+  /**
+   * **Weak hit:** The score beats one challenge die.
+   */
+  'Weak Hit' = 1,
+  /**
+   * **Strong hit:** The score beats both challenge dice.
+   */
+  'Strong Hit' = 2,
 }
 
+// TODO: consider differentiating `source` from a new prop called e.g. `label`; it's pulling double duty now as both as something being tested for internal logic *and* as a user-facing label. even if those are mutually exclusive categories right now, it might be risky in the long term to conflate them.
```

Comment:
Is this true though? I don't see any places where it's used as anything but an annotation that will eventually make it to the UI.

Reply:
my thinking here is: "thing that is checked by internal logic" and "think that gets provided to the user as a label" are very different purposes with different needs, which i'd tend towards making their own keys (having been bitten by "eh, it's *pretty close* in the past" on dataforged, lol). *right now* they overlap enough that it's no big deal, but if those needs expand (e.g. by expanding localization), that might change.

i reckon part of what makes me nervous is relying on various strings being changed (or, to frame it using the "different purposes/needs" above: their purpose/needs being changed midstream). it strikes me as something that could be a bit fragile, especially with locale strings (and user-provided strings e.g. labels drawing from custom asset meter names) in the mix.

YMMV on that, though. FWIW, if i were trying to address some of the above but keep it down to 2 props, i might structure it something like:

* keep **anything** that could be user input out of `source` entirely
* avoid localized i18n strings; if *absolutely* necessary, use their key instead 
* where possible, enumerate valid `source` values with an enum (it might then be typed as a union of those enums)
* a more elaborate version might also include:
  * FVTT ids, perhaps prefixed with item type using a reliable delimiter e.g. `item:dkhqeleflq1255r`
    - include a way to reference their properties e.g. `item:dkhqeleflq1255r#data.whatever`, to allow for e.g. referencing the label of a custom asset meter

but all that actually seems like more work (and is probably less flexible than) than just adding an optional third prop, where it's safe to put any old string because it'll never be parsed.

Reply:
wow, that was a lot of words! anyways -- the answer to all of that might be simply: "its unlikely to cause problems, so it's good enough for now so it gets into people's hands and get feedback to iterate on". fair enough -- just want to register my concerns before it's released into the wild ;)

Reply:
actually: i've just run into a situation where this makes it difficult to make even minor adjustments to string composition, spotted as i was trying to restore proper casing to e.g. stat names.
<img width="215" alt="Screen Shot 2022-08-23 at 2 27 36 AM" src="https://user-images.githubusercontent.com/5354757/186123453-a701a5ab-a70e-45dc-8378-df7477f9d4c1.png">
doesn't seem optimal to have the name there twice. ideally, it would only show "(Marc Progress)" if the message was being attributed to someone other than Marc Progress.

sounds simple enough. but the `source` string being a prebuilt label, "health (Marc Progress)", means i have to write something to parse the name back out of the string... and that all assumes it's the same actor in the first place (they might just have the same name).

and god help me if i try to do that with an actor.data.name that includes any parentheses. most names don't. but there's some not-uncommon cases for that, just off the top of my head:

* in FVTT specifically, i've seen it used to denote PC ownership ("Charname (playername)") or pronouns ("Charname (he/him)") of a character in ways that FVTT's UX doesn't address (e.g.:  to indicate 'ownership' of a character in a game where everyone is a GM, or if they want them to be easily available without opening the character sheet)
* in multilingual settings in the real world, someone might provide the name they use in one languge, and then in parentheses provide the name they use in another language. in vancouver i see this pretty regularly: someone's english name is followed by their name's original rendering as hanzi characters wrapped in parentheses

Reply:
Okay, what if instead of adding another field to `SourcedValue`, we instead add a parameter to `PreRollOptions` to indicate the "title" or "name" or "caption" for the roll? I think we both get what we want there; the `source` field stays as (mostly) hard-coded-through-i18n hints that (mostly) end up in tooltips, and we can still augment the roll title for asset tracks.

Reply:
yeah, `title` makes a lot of sense to me - maybe as an optional that falls back to its present value? then it's a bit easier to change incrementally